addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -0,0 +1,397 @@
+                                break;
+                            case SOURCE_INTERNET_RADIO:
+                                if (!sourceName.isEmpty()) {
+                                    String id = xmlHelper.parsePresets(XmlHelper.REST_CURRENT_PRESET, response,
+                                            sourceName);
+                                    if (!id.isEmpty()) {
+                                        refreshChannels(OnOffType.ON, new StringType(SOURCE_PRESET_PREFIX + id),
+                                                sourceName, PlayPauseType.PLAY);
+                                    }
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                }
+                if (command instanceof StringType) {
+                    StringType strintType = (StringType) command;
+                    switch (strintType.toString()) {
+                        case SOURCE_PRESET_1:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_1);
+                            break;
+                        case SOURCE_PRESET_2:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_2);
+                            break;
+                        case SOURCE_PRESET_3:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_3);
+                            break;
+                        case SOURCE_PRESET_4:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_4);
+                            break;
+                        case SOURCE_PRESET_5:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_5);
+                            break;
+                        case SOURCE_PRESET_6:
+                            response = wsHelper.pressAndReleaseButtonOnSpeaker(BSTKeys.PRESET_6);
+                            break;
+                        case SOURCE_AUX:
+                            response = wsHelper.selectAUX();
+                            break;
+                        case SOURCE_BLUETOOTH:
+                            response = wsHelper.selectBluetooth();
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+        checkResponseForError(response);
+    }
+
+    private void checkResponseForError(String response) {
+        if (response.startsWith(ERROR_STRING)) {
+            socketListener = null;
+            try {
+                if (client != null) {
+                    client.stop();
+                }
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception